### PR TITLE
spec source for cl_khr_kernel_clock

### PR DESCRIPTION
**Pull request by bashbaug**

### CODE REVIEW

**Comment by bashbaug** [comment]
TODOS:

- Clarify that this is a provisional extension.
- Remove "ext" from feature test macros.

**Comment by bashbaug** [comment]
I've fixed all of the TODOs listed above.  I think this is ready to go.

File: api/cl_khr_kernel_clock.asciidoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+=== New Enums
+
+  * {cl_device_info_TYPE}
+  ** {CL_DEVICE_KERNEL_CLOCK_CAPABILITIES_KHR}
+  * {cl_device_kernel_clock_capabilities_khr_TYPE}
+  ** {CL_DEVICE_KERNEL_CLOCK_SCOPE_DEVICE_KHR}
+  ** {CL_DEVICE_KERNEL_CLOCK_SCOPE_WORK_GROUP_KHR}
+  ** {CL_DEVICE_KERNEL_CLOCK_SCOPE_SUB_GROUP_KHR}
+
+include::provisional_notice.asciidoc[]
```
Comment (oddhack) [inline api/cl_khr_kernel_clock.asciidoc:59]: Should be removed if #1113 is accepted.

File: api/cl_khr_kernel_clock.asciidoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+=== Version History
+
+  * Revision 1.0.0, 2024-03-25
+  ** Initial version
```
Comment (oddhack) [inline api/cl_khr_kernel_clock.asciidoc:64]: Should say something about being "(provisional)"? Other extension version histories do.

**Comment by oddhack** [comment]
Aside from minor comments, LGTM from a markup perspective. I am a little confused though, because I thought this extension was in the recently ratified as non-provisional list that Neil sent out (and was thus surprised not to see an appendix for it when constructing #1113).

**Comment by oddhack** [comment]
> (and was thus surprised not to see an appendix for it when constructing #1113).

To be clear, I am looking at the minutes from March 19th which have a "Ratification #333 - The following extensions are now ratified as final specifications" section including cl_khr_kernel_clock, so I assumed this had previously been published as provisional and wondered where the appendix was. I haven't looked into it any more deeply than that.

File: api/cl_khr_kernel_clock.asciidoc (hunk 1 above)
Comment (bashbaug) [reply api/cl_khr_kernel_clock.asciidoc:59]: Fixed.

File: api/cl_khr_kernel_clock.asciidoc (hunk 2 above)
Comment (bashbaug) [reply api/cl_khr_kernel_clock.asciidoc:64]: Fixed.

**Comment by bashbaug** [comment]
> I am a little confused though, because I thought this extension was in the recently ratified as non-provisional list that Neil sent out

Yeah, I was initially confused by this too.  The change to a provisional ratification was done at the last minute.

The ratification review package clearly says provisional though, so I believe this is correct.  Regardless, I think it would be easier to change from provisional to final than final to provisional.

**Comment by bashbaug** [comment]
Agreed to merge in principle after editorial review.

**Review by alycm — APPROVED** [review]
LGTM

**Comment by bashbaug** [comment]
Thanks, merging.  Please file a GitHub issue if any other issues are found, editorial or otherwise.

**Comment by bcalidas** [comment]
Extension text looks good.

**Pull request merged**